EmeraldComponents/Basics/Label/EmeraldLabel.swift

```diff
@@ -0,0 +1,305 @@
+    
+    internal private(set) var type: EmeraldLabelType = .fill {
+        didSet {
+            self.typeOrStyleDidChange(self.state, self.type)
+        }
+    }
+
+    internal private(set) var state: EmeraldLabelState = .error {
+        didSet {
+            self.typeOrStyleDidChange(self.state, self.type)
+        }
+    }
+    
+    // MARK: - Aux properties
+    
+    /// Used to set a default value for icon image
+    internal var auxIconLabel: String = "icon-dot"
+    
+    /// Used to set a default value for text label
+    internal var auxTextLabel: String = "Label"
+    
+    // MARK: - Init
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.xibSetup()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.xibSetup()
+    }
+    
+    private func xibSetup() {
+        Bundle.basic.loadNibNamed(String(describing: EmeraldLabel.self), owner: self, options: nil)
+        
+        addSubview(self.contentView)
+        
+        self.contentView.frame = self.bounds
+        self.contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        
+        self.setDefaultValuesToLabels()
+    }
+    
+    internal func setDefaultValuesToLabels() {
+        self.textLabel.layer.cornerRadius = 15.0
+        self.textLabel.layer.masksToBounds = true
+        self.textLabel.font = UIFont.systemFont(ofSize: 15.0, weight: .heavy)
+    }
+    
+    // MARK: - Public configuration
+    
+    /// Set an icon that will be displayed to the left of label.
+    ///
+    /// - Parameter name: The name of an image inside emerald project assets. You can't choose any image. Only emerald images
+    public func setIcon(name: String) {
+        self.auxIconLabel = name
+    }
+    
+    /// Set an label text that will be displayed to user.
+    ///
+    /// - Parameter Text: The text label name.
+    public func setLabel(text: String) {
+        self.auxTextLabel = text.uppercased()
+    }
+    
+    /// Sets the status and type
+    ///
+    /// - Parameters:
+    ///   - state: Label state to be displayed
+    ///   - type: Label type to be displayed
+    public func configure(state: EmeraldLabelState, type: EmeraldLabelType) {
+        self.state = state
+        self.type = type
+    }
+    
+    // MARK: - State changes
+
+    /// The main method responsible to get the current state and type and set it to 'EmeraldLabel.xib' properties
+    ///
+    /// - Parameters:
+    ///   - state: Label state enum
+    ///   - type: Label type enum
+    internal func typeOrStyleDidChange(_ newState: EmeraldLabelState, _ newType: EmeraldLabelType) {
+        self.textLabel.text = self.text(forState: newState, andType: newType)
+        self.textLabel.backgroundColor = self.backgroundColor(forState: newState, andType: newType)
+        self.textLabel.textColor = textColor(forState: newState, andType: newType)
```

Comment:
```suggestion
        self.textLabel.textColor = self.textColor(forState: newState, andType: newType)
```
Just to keep the pattern? (TOC 😂)